src/ifcopenshell-python/ifcopenshell/validate.py

```diff
@@ -563,9 +564,31 @@ def validate_guid(guid: str) -> Union[str, None]:
         return "Couldn't decompress guid, it's not base64 encoded."
     return None
 
+def to_string_header_entity(header_entity):
+    """Recreate IFC header string representation, like FILE_NAME(...)"""
+    def format_value(val):
+        if isinstance(val, str):
+            return f"'{val}'"
+        elif isinstance(val, tuple):
+            return "(" + ",".join(format_value(v) for v in val) + ")"
+        return str(val)
+
+    # Prefer native .toString() if available (native IfcOpenShell wrapper)
+    if hasattr(header_entity, 'toString'):
+        return header_entity.toString()
+
+    if hasattr(header_entity, '_fields'):
+        values = [format_value(getattr(header_entity, f)) for f in header_entity._fields]
+    else:
+        raise TypeError(f"Cannot stringify header_entity of type {type(header_entity)}")
+
+    return f"{type(header_entity).__name__.upper()}({','.join(values)})"
 
-def validate_ifc_header(f: ifcopenshell.file, logger: Logger) -> None:
-    header: W.IfcSpfHeader = f.wrapped_data.header
+def validate_ifc_header(f: Union[ifcopenshell.file, ifcopenshell.simple_spf.file], logger: Logger) -> None:
+    if type(f) is ifcopenshell.file:
+        header: W.IfcSpfHeader = f.wrapped_data.header
+    else:
+        header: types.SimpleNamespace = f.header
```

Comment:
```suggestion
    header: Union[W.IfcSpfHeader, types.SimpleNamespace] = f.header
```

They are fully compatible, no need for the if statement

Reply:
[Done](https://github.com/IfcOpenShell/IfcOpenShell/pull/6650/commits/cb093001529a8d4d087d696ee16f0d0bdf70a1e7)